src/plasmapy/plasma/material.py

```diff
@@ -0,0 +1,193 @@
+"""
+Definition of the material class and helper functions. The module also
+includes type definitions for stopping and ranging data.
+"""
+
+from collections.abc import Callable
+from functools import cached_property
+from typing import TypedDict
+
+import astropy.units as u
+import numpy as np
+from scipy.interpolate import CubicSpline
+
+from plasmapy.particles.particle_class import Particle
+
+__all__ = [
+    "MaterialSTARData",
+    "Material",
+]
+
+from plasmapy.utils.decorators import validate_quantities
+
+
+class MaterialSTARData(TypedDict):
+    """Type definition for formatting data related to beam-target interactions."""
+
+    incident_particle: Particle
+
+    energies: u.Quantity[u.MeV]
+    stopping_power: u.Quantity[
+        u.MeV * u.cm**2 / u.g
+    ]  # Units of energy per areal density
+    CSDA_range: u.Quantity[u.cm]
+    projected_range: u.Quantity[u.cm]
+
+
+def _construct_log_interpolator(
+    x0: u.Quantity, y0: u.Quantity
+) -> Callable[[u.Quantity], u.Quantity]:
+    """
+    Apply a logarithmic transformation to input and output data points,
+    allowing for approximation of an exponential relation using
+    a cubic spline.
+    """
+    x_unit = x0.unit
+    y_unit = y0.unit
+
+    log_interpolator = CubicSpline(
+        x=np.log(x0.value),
+        y=np.log(y0.value),
+    )
+
+    def interpolator(x_unsanitized: u.Quantity[x_unit]) -> u.Quantity[y_unit]:
+        x = x_unsanitized.to(x_unit).value
+        log_x = np.log(x)
+
+        return np.exp(log_interpolator(log_x)) * y_unit
+
+    return interpolator
+
+
+class Material:
+    """A representation of a compound material.
+
+    Parameters
+    ----------
+    A : `~astropy.units.Quantity`
+        An array of the atomic weights of the elements in the compound in units
+        convertible to atomic mass units.
+
+    Z : `~astropy.units.Quantity`
```

Comment:
If A is the atomic number, then Z is usually the ionization state in a plasma. But this is a solid, so I think you only need A. In the Moliere scattering then, "z" is "A".